### PR TITLE
dev/core#905 Separate contribution_recur status_id option group from contribution option group

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
No user impact, splits contribution recur status option group from contribution

Before
----------------------------------------
One option group shared

After
----------------------------------------
Two separate ones

Technical Details
----------------------------------------
As we did earlier for pledge status this separates out the option groups - on the basis we have different statuses for the 2 things. In order to ensure we don't break anything 
we give the new option group the exact same values as the old one

For new sites I have left off some statuses that I think would not be relevant - Refunded,
Partially Paid, Chargeback, Pending Refund

Comments
----------------------------------------
@mattwire @monishdeb @pradpnayak @adixon input welcome!

A couple of follow up notes

1) I think 'In Progress' is now not needed in the contribution_status option group & we might remove from new installs
2) @mattwire this also solves a problem I hit when trying to make the contribution recur search form metadata driven. 
3) Regarding adding the statuses to add - there were some comments on this PR template but I've removed them from here as the discussion is in gitlab & I will add a follow up PR with them

https://lab.civicrm.org/dev/core/issues/905

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton don't we need the updated generated_data.sql?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 good question - I guess I run setup.sh?

File: xml/schema/Contribute/ContributionRecur.xml [hunk 1]
```diff
@@ -241,7 +241,7 @@
     <import>true</import>
     <add>1.6</add>
     <pseudoconstant>
-      <optionGroupName>contribution_status</optionGroupName>
+      <optionGroupName>contribution_recur_status</optionGroupName>
```
Comment (seamuslee001) [inline xml/schema/Contribute/ContributionRecur.xml]: need to run `./bin/setup -g` to update the DAO

**Comment by eileenmcnaughton** [comment]
Cool done.

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by mattwire** [comment]
@eileenmcnaughton Thoughts added to the gitlab issue.  Summary: Yes, but we need more thought on the options we provide before merging.

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@mattwire I've altered this to NOT add the new status. I feel like this would be better merged as a PR that just adds the group & I can follow up & add the statuses. I think we have agreed on them but it feels like a separate PR would be cleaner

**Comment by adixon** [comment]
Good idea, for reasons already well noted, and I support the implementation strategy. 

I'm only confused about the note about removing the "In Progress" status - are you talking about removing that from the non-recurring payment status?

**Comment by eileenmcnaughton** [comment]
@adixon when I wrote that I believed that 'In Progress' was relevant to pledges & recurring contributions - but not actual contributions. So I thought that with neither of those entities sharing the contribution status option group anymore it would not be needed in that group (I have seem it cause some confusion). However after doing some other stuff in BAO_Contribution yesterday I realise it IS referenced in there for contribution handling so I think I was wrong

File: CRM/Upgrade/Incremental/sql/5.15.alpha1.mysql.tpl [hunk 2]
```diff
@@ -1 +1,11 @@
 {* file to handle db changes in 5.15.alpha1 during upgrade *}
+--dev/core#905 Add contribution recur status option group
+INSERT INTO `civicrm_option_group`  ( `name`, {localize field='title'}`title`{/localize}, `is_active` ) VALUES ('contribution_recur_status', {localize}'{ts escape="sql"}Contribution Recur Status{/ts}'{/localize}, 1);
```
Comment (pradpnayak) [inline CRM/Upgrade/Incremental/sql/5.15.alpha1.mysql.tpl]: Contribution status option group has is_reserved and is_locked to TRUE, should this option should have same value for those fields?

Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/sql/5.15.alpha1.mysql.tpl]: @pradpnayak yes, thank you - I will fix!

File: xml/templates/civicrm_data.tpl [hunk 3]
```diff
@@ -212,6 +212,7 @@ VALUES
    ('wysiwyg_presets'               , '{ts escape="sql"}WYSIWYG Editor Presets{/ts}'             , NULL, 1, 1, 0),
    ('relative_date_filters'         , '{ts escape="sql"}Relative Date Filters{/ts}'              , NULL, 1, 1, 0),
    ('pledge_status'                 , '{ts escape="sql"}Pledge Status{/ts}'                      , NULL, 1, 1, 1),
+   ('contribution_recur_status'     , '{ts escape="sql"}Recurring Contribution Status{/ts}'      , NULL, 1, 1, 1),
```
Comment (pradpnayak) [inline xml/templates/civicrm_data.tpl]: Label is different here v/s in upgrade script.

**Comment by eileenmcnaughton** [comment]
@pradpnayak thank you ! I think I've fixed those now

**Comment by pradpnayak** [comment]
Tested this PR on fresh install and upgrade. 
On fresh install - New recurring contribution status option group is created with options

On upgrade - New recurring contribution status option group is created with options of Contribution status copied over.

The options doesn't match on Fresh install v/s upgrade since on fresh install it creates 6 options while during upgrade it creates 10 options(by default contribution statuses). This does make sense inorder to keep the consistency of option.value.

Good to merge.

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by yashodha** [comment]
@eileenmcnaughton : the tests have all passed.
Can this be merged or does it need further inputs/discussions?

**Comment by eileenmcnaughton** [comment]
@yashodha I think it's good to merge - I kept all the stuff that might need discussion for a follow up PR - which I'll do once this is merged

**Pull request merged**